core/src/main/java/io/ddf/etl/TransformationHandler.java

```diff
@@ -106,8 +106,37 @@ public DDF flattenDDF() {
     return flattenDDF(null);
   }
 
-  public DDF transformUDF(String RExprs, List<String> columns)
-          throws DDFException {
+
+  public DDF flattenArrayTypeColumn(String colName) throws DDFException {
+    StringBuffer newCols = new StringBuffer();
+    Column arrCol = this.getDDF().getColumn(colName);
+    if (arrCol.getType() != ColumnType.ARRAY) {
+      throw new DDFException("Column to be flattened must be an array");
+    }
+    String arrSizeStr = this
+        .getDDF()
+        .sql(String.format("SELECT size(%s) FROM @this LIMIT 1", colName),
+            "Unable to fetch the first value of the requested column").getRows().get(0);
+
+    int arrSize = Integer.parseInt(arrSizeStr);
+    for (int i = 0; i < arrSize; i++) {
+
+      newCols.append(String.format(" %s[%d] as %s_c%d,", colName, i, colName, i));
+    }
+
+    newCols.setLength(newCols.length() - 1);
+    String sqlCmd = String.format("SELECT *, %s FROM @this", newCols.toString());
+    DDF newddf = this.getDDF().sql2ddf(sqlCmd);
+
+    if (this.getDDF().isMutable()) {
```

Comment:
@Huandao0812 @binhmop 
We don't have mutable logic anymore, all DDF should be mutable by default. Should we remove this unnecessary logic?